Lib/encodings/idna.py

```diff
@@ -156,7 +170,7 @@ def encode(self, input, errors='strict'):
 
         if errors != 'strict':
             # IDNA is quite clear that implementations must be strict
-            raise UnicodeError("unsupported error handling "+errors)
+            raise UnicodeEncodeError("idna", input, 0, 1, f"unsupported error handling {errors}")
```

Comment:
```suggestion
            raise UnicodeEncodeError("idna", input, 0, 0, f"unsupported error handling {errors}")
```

Reply:
UnicodeEncodeError implies that there is an error in the encoded string.

Other codecs usually raise LookupError for unknown error handlers, but here an exception is raised even for known error handlers.

Perhaps ValueError better suits here. UnicodeError is a subclass of ValueError, so we can keep it for now.

Reply:
Changed the cases where there was not an issue with the input string back to `UnicodeError`.